apps/web/components/auth/nextauth-layout-wrapper.tsx

```diff
@@ -39,7 +42,7 @@ export const NextAuthLayoutWrapper: FC<NextAuthLayoutWrapperProps> = ({
       // Redirect to legal acceptance if not accepted
       router.push("/legal-acceptance");
     }
-  }, [session, status, router, legalStatus, pathname]);
+  }, [session, status, router, legalStatus, pathname, searchParams]);
```

Comment:
searchParams is likely the same object, even if the searchParams change. Just for correctness, calculate `search` outside the effect, and then make the effect dependent on `search`